File: flutter/lib/features/champion/domain/usecase/get_champion.dart
import 'package:lol_champion/features/champion/domain/entity/champion.dart';
import 'package:lol_champion/features/champion/domain/repository/champion_repository.dart';

class GetChampion {
  const GetChampion(this._repository);

  final ChampionRepository _repository;

  Future<Champion> call(String championId) =>
      _repository.getChampion(championId);
}

File: react/src/features/champion/domain/use-cases/get-champion.ts
import type { Champion } from '@/features/champion/domain/entities/champion';
import type { ChampionRepository } from '@/features/champion/domain/repositories/champion-repository';

export class GetChampion {
  constructor(private readonly repository: ChampionRepository) {}

  execute(championId: string, patchVersion: string, locale: string): Promise<Champion> {
    return this.repository.getChampion(championId, patchVersion, locale);
  }
}
